src/pages/SingleBreedPage/SingleBreed.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect } from "react";
+import "./singleBreed.scss";
+import { useLocation } from "react-router-dom";
+import { BreedType } from "../../types";
+import { Link } from "react-router-dom";
+import loadingImg from "../../assets/loading.gif";
+
+export default function SingleBreed() {
+  let { state } = useLocation();
+  const [breed, setInfo] = useState<BreedType[]>([]);
+  const [errorMessage, setError] = useState<string>("");
+  let dog = breed.find((item) => item.id === state.id) as BreedType;
+  let dogDetails = breed.length > 0 ? Object.keys(dog).slice(3, -2) : [];
+
+  useEffect(() => {
+    fetch("https://api.thedogapi.com/v1/breeds?limit=20&page=0")
```

Comment:
You don't have to fetch all dogs. You must fetch only one dog by URL like `https://api.thedogapi.com/v1/breeds/1`